proposals/lip-0045.md

```diff
@@ -1575,6 +1446,47 @@ genesisInteroperabilityStoreSchema = {
 
 Here, the `...` notation, borrowed from [JavaScript ES6 data destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring), indicates that the corresponding schema should be inserted in place, and it is just used for notational convenience.
 
+#### Genesis State Initialization
+
+During the genesis state initialization stage of a genesis block `g`, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the genesis block assets for the Interoperability module and let `interoperabilityAsset = decode(genesisInteroperabilityStoreSchema, genesisBlockAssetBytes)`. Let `chainInfos = interoperabilityAsset.chainInfos`.
+
+##### Genesis Asset Verification
+
+* Within `chainInfos`, check that all entries have a different `chainID`.
+* For every `chainInfo` in `chainInfos`, properties `chainInfo.chainData`, `chainInfo.channelData` and `chainInfo.chainValidators` must be non-empty.
```

Comment:
Here maybe we need to be a bit more precise. In my understanding, these properties are objects, so that 'empty' means that all properties are set to their default values. So I am wondering if this check is not actually redundant: for instance, the default value of the chain data would be invalid against the schema validation.

Reply:
I split the properties into several arrays, to avoid this problem